stream/stream.go

```diff
@@ -0,0 +1,208 @@
+package stream
+
+type ColourStop struct {
+	Offset string // e.g. "0%", "50%", "100%"
+	Color  string // e.g. "#ff0000"
+}
+
+type Stream struct {
+	// key is the identifier and doubles as the name (probably a terrible idea, but it hasn't bitten me yet).
+	key string
+	// description is just some more info about this stream.
+	description string
+	// unit of the values in points.
+	unit string
+	// discrete determines whether to treat data points as discrete which essentially means when they change there is no
+	// "in between" values. An example is gears; selected gear could be 1 or 2, but can't be 1.5.
+	discrete bool
+	// smoothingAlpha determines how much smoothing to apply, 1 is no smoothing and very responsive, 0 is lots of
+	// smoothing and less responsive. This uses EMA smoothing.
+	smoothingAlpha float64
+	// precision determines how many decimal places to show when displaying the value.
+	precision uint8
+	// colours is an array of colours to use if this data is displayed, it is treated as a gradient where low values
+	// are given the first colour in the slice, and high values are given the last colour in the slice. Colours should
+	// be specified as 3 byte hex without the #.
+	colours []ColourStop
+	// min is the minimum value to show on the y-axis
+	min float64
+	// max is the maximum value to show on the y-axis
+	max float64
+	// windowSize determines how many milliseconds worth of data to show.
+	windowSize int
+	// IsActive determines whether this stream is the active stream within it's chart
+	IsActive bool
+	// points holds the actual data.
+	points []DataPoint
+	// window holds point data, post processed from points.
+	window []DataPoint
+}
+
+func NewStream(
```

Comment:
Is there much purpose for a NewStream function if there aren't any defaults applied and it expects a param per attribute anyway? 🤔

Reply:
Na not really lol, I had grand plans but it wasn't super necessary

Reply:
Actually I like constructors cause it lets me know exactly what is required to instantiate a struct. ie; in the future I might not know that the two arrays of data points need to be created and should be empty. I guess it does have defaults.